bootstrap/v2/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -577,84 +572,392 @@ func (kustomize *kustomize) writeConfigFile() error {
+			for _, fileSource := range childConfigMapArgs.FileSources {
+				fileAbsolutePathSource := path.Join(targetDir, fileSource)
+				parentConfigMapArgs.EnvSource = extractSuffix(compDir, fileAbsolutePathSource)
+			}
+		}
+		if childConfigMapArgs.LiteralSources != nil && len(childConfigMapArgs.LiteralSources) > 0 {
+			parentConfigMapArgs.LiteralSources = make([]string,0)
+			for _, literalSource := range childConfigMapArgs.LiteralSources {
+				parentConfigMapArgs.LiteralSources = append(parentConfigMapArgs.LiteralSources, literalSource)
+			}
+		}
+		switch types.NewGenerationBehavior(childConfigMapArgs.Behavior) {
+		case types.BehaviorCreate:
+			if _, ok := kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name]; !ok {
+				parent.ConfigMapGenerator = append(parent.ConfigMapGenerator, *parentConfigMapArgs)
+				kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name] = true
+			}
+		case types.BehaviorMerge, types.BehaviorReplace, types.BehaviorUnspecified:
+			fallthrough
+		default:
+			parent.ConfigMapGenerator = append(parent.ConfigMapGenerator, *parentConfigMapArgs)
+			kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name] = true
+		}
+	}
+
+	updateParamFiles()
+	if child.Bases == nil {
+		basePath := extractSuffix(compDir, targetDir)
+		if _, ok := kustomizationMaps[basesMap][basePath]; !ok {
+			parent.Bases = append(parent.Bases, basePath)
+			kustomizationMaps[basesMap][basePath] = true
+		}
+		return nil
+	}
+	for _, value := range child.Bases {
+		baseAbsolutePath := path.Join(targetDir, value)
+		basePath := extractSuffix(compDir, baseAbsolutePath)
+		if _, ok := kustomizationMaps[basesMap][basePath]; !ok {
+			parent.Bases = append(parent.Bases, basePath)
+			kustomizationMaps[basesMap][basePath] = true
+		} else {
+			childPath := extractSuffix(compDir, targetDir)
+			kustomizationMaps[basesMap][childPath] = true
+		}
+	}
+	if child.NamePrefix != "" {
+		log.Warnf("cannot merge nameprefix %v ", child.NamePrefix)
+	}
+	if child.GeneratorOptions != nil && parent.GeneratorOptions == nil {
+		parent.GeneratorOptions = child.GeneratorOptions
+	}
+	if child.NameSuffix != "" {
+		log.Warnf("cannot merge namesuffix %v ", child.NamePrefix)
+
+	}
+	if len(child.CommonLabels) > 0 {
+		log.Warnf("cannot merge commonlabels %v ", child.CommonLabels)
+
+	}
+	if len(child.CommonAnnotations) > 0 {
+		log.Warnf("cannot merge commonannotations %v ", child.CommonAnnotations)
+	}
+	for _, value := range child.Resources {
+		resourceAbsoluteFile := filepath.Join(targetDir, string(value))
+		resourceFile := extractSuffix(compDir, resourceAbsoluteFile)
+		if _, ok := kustomizationMaps[resourcesMap][resourceFile]; !ok {
+			parent.Resources = append(parent.Resources, resourceFile)
+			kustomizationMaps[resourcesMap][resourceFile] = true
+		}
+	}
+	for _, value := range child.Images {
+		imageName := value.Name
+		if _, ok := kustomizationMaps[imagesMap][imageName]; !ok {
```

Comment:
if multiple overlays define same imageName, always use the first overlay. Should we log a msg about ignoring later images? so do other fields, such as crd, Vars